apiserver/facades/controller/caasapplicationprovisioner/provisioner.go

```diff
@@ -252,14 +254,22 @@ func (a *API) watchProvisioningInfo(appName names.ApplicationTag) (params.Notify
 	controllerAPIHostPortsWatcher := a.ctrlSt.WatchAPIHostPortsForAgents()
 	modelConfigWatcher := model.WatchForModelConfigChanges()
 
-	multiWatcher := common.NewMultiNotifyWatcher(appWatcher, controllerConfigWatcher, controllerAPIHostPortsWatcher, modelConfigWatcher)
+	multiWatcher, err := eventsource.NewMultiNotifyWatcher(ctx,
+		appWatcher,
```

Comment:
I know these are coming from different sources including deprecated state, but I envisage a future where these are just single watchers based on a filter.

Reply:
That's the dream.